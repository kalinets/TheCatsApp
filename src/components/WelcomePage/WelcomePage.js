// @flow

import React from 'react'
import { Link } from 'react-router-dom'
import AuthUserContext from '../AuthUserContext/AuthUserContext'

const WelcomePage = () => (
  <>
    <h2>Welcome to cat lovers app.</h2>
    <AuthUserContext.Consumer>
      {user => (
        <>
          {!user && (
            <p>
              <Link to="/signup">Create account</Link>
              {` to see beautiful cats and have fun ;)`}
            </p>
          )}
        </>
      )}
    </AuthUserContext.Consumer>
  </>
)

export default WelcomePage
