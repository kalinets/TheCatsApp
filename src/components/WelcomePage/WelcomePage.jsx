import React from 'react'
import { Link } from 'react-router-dom'

const WelcomePage = () => (
  <>
    <h2>Welcome to cat lovers app.</h2>
    <p>
      <Link to="/signup">Create an account</Link> if you have not one.
    </p>
  </>
)

export default WelcomePage
