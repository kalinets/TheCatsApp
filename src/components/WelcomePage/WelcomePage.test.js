import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import { shallow } from 'enzyme'
import WelcomePage from './WelcomePage'

configure({ adapter: new Adapter() })

describe('<WelcomePage />', () => {
  test('renders without crashing', () => {
    shallow(<WelcomePage />)
  })
  test('matches snapshot', () => {
    expect(shallow(<WelcomePage />)).toMatchSnapshot()
  })
})
