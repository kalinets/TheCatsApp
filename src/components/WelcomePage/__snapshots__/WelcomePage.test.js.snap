// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomePage /> matches snapshot 1`] = `
<Fragment>
  <h2>
    Welcome to cat lovers app.
  </h2>
  <ContextConsumer>
    <Component />
  </ContextConsumer>
</Fragment>
`;
